service.go

```diff
@@ -282,7 +288,14 @@ func NewService(serviceName string, serviceVersion string, confProfile string, c
 	svc = &Service{}
 	svc.startTime = startTime
 	svc.svcInfo = &config.Service
-	common.Driver = proto
+	if casted, ok := proto.(dsModels.ProtocolDriver); ok {
+		common.Driver = casted
+	}
+	if casted, ok := proto.(dsModels.ProtocolDiscovery); ok {
+		common.Discovery = casted
+	} else {
+		common.Discovery = nil
+	}
 
```

Comment:
You have to deal with the case if proto is neither dsModels.ProtocolDriver nor dsModels.ProtocolDiscovery